### PR TITLE
Add SelectionSort and BubbleSort

### PR DESCRIPTION
**Pull request by Sunbelife**
I added the BubbleSort, Selection and their benchmarks.

Signed-off-by: Sunbelife Sunbelife@foxmail.com

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/Llama-Group/Project-Llama/pull/59" target="_blank">ReviewNinja

File: llama_cpp/src/Sort/SelectionSort.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+		void doSort(std::vector<SortObject<T>> *reformedArray) override {
+			SortObject<T> temp;
+			for (int i = 0; i < reformedArray->size(); i++) {
+				int flag = i;
+				SortObject<T> key = reformedArray->at(i);
+				//select the min value
+				for (int j = i; j < reformedArray->size(); j++){
+					if (key > reformedArray->at(j))
+					{
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:24]: Style

File: llama_cpp/src/Sort/SelectionSort.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		SelectionSort() {}
+		SelectionSort(std::vector<T> *inputArray) { this->sort(inputArray); }
+
+		void doSort(std::vector<SortObject<T>> *reformedArray) override {
+			SortObject<T> temp;
+			for (int i = 0; i < reformedArray->size(); i++) {
+				int flag = i;
+				SortObject<T> key = reformedArray->at(i);
+				//select the min value
+				for (int j = i; j < reformedArray->size(); j++){
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:22]: Style, missing space

File: llama_cpp/src/Sort/SelectionSort.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	public:
+		SelectionSort() {}
+		SelectionSort(std::vector<T> *inputArray) { this->sort(inputArray); }
+
+		void doSort(std::vector<SortObject<T>> *reformedArray) override {
+			SortObject<T> temp;
+			for (int i = 0; i < reformedArray->size(); i++) {
+				int flag = i;
+				SortObject<T> key = reformedArray->at(i);
+				//select the min value
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:21]: Comment style

File: llama_cpp/src/Sort/SelectionSort.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+				for (int j = i; j < reformedArray->size(); j++){
+					if (key > reformedArray->at(j))
+					{
+						flag = j;
+					}
+				}
+				//swap between flag and the i
```
Comment (0xsuu) [inline llama_cpp/src/Sort/SelectionSort.h:28]: Comment style

File: llama_cpp/src/Sort/BubbleSort.h [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	public:
+		BubbleSort() {}
+		BubbleSort(std::vector<T> *inputArray) { this->sort(inputArray); }
+
+		void doSort(std::vector<SortObject<T>> *reformedArray) override {
+			SortObject<T> temp;
+			for (int i = 0; i < reformedArray->size()-1; i++) {
+				for (int j = 0; j < reformedArray->size()-1-i; j++) {
+					if(reformedArray->at(j) > reformedArray->at(j+1))
+					//swap between i+1 and the i
```
Comment (0xsuu) [inline llama_cpp/src/Sort/BubbleSort.h:21]: Comment style

File: llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp [hunk 6]
```diff
@@ -67,6 +69,55 @@ static void InSortDouble(benchmark::State& state) {
         inSort.sort(&a);
```
Comment (0xsuu) [inline llama_cpp/benchmark/sort_benchmark/sort_benchmark.cpp:69]: InSort not inSort

File: llama_cpp/example/sort_test/sort_test.cpp [hunk 7]
```diff
@@ -91,5 +93,45 @@ int main() {
 		std::cout << std::string(d[i]) << " " << std::endl;
 	}
 
+	// SelectionSort
```
Comment (0xsuu) [inline llama_cpp/example/sort_test/sort_test.cpp:96]: Should be a sentence.

File: llama_cpp/example/sort_test/sort_test.cpp [hunk 8]
```diff
@@ -91,5 +93,45 @@ int main() {
...
+
+	std::cout << "SelectionSort results:" << std::endl;
+	for (int i = 0; i < 5; i++) {
+		std::cout << a[i] << " ";
+		std::cout << b[i] << " ";
+		std::cout << c[i] << " ";
+		std::cout << std::string(d[i]) << " " << std::endl;
+	}
+
+	// BubbleSort
```
Comment (0xsuu) [inline llama_cpp/example/sort_test/sort_test.cpp:116]: Should be a sentence.

**Comment by 0xsuu** [comment]
@Sunbelife DON'T CREATE NEW PR

**Pull request closed without merging**